### PR TITLE
chore(nix): updates to use nixpkgs elixir 1.17

### PR DESCRIPTION
**Pull request by adamcstephens**
I split the package out here, and dropped the makeOverridable. I can put this back if needed, but I think separating the package makes it clearer to reason about. It's also more similar to a nixpkgs package this way. :)

I discovered that erlang_27 actually depends on erlang_26 currently for ex_doc, so you have to compile both with beam_minimal. Anybody using your cache won't notice though.

I also wondered if the zig pinned release could be removed in favor of current unstable.

### CODE REVIEW

File: flake.nix
Comment (mhanberg) [inline flake.nix:26]: i don't think we need to extend this anymore

**Comment by mhanberg** [comment]
> I also wondered if the zig pinned release could be removed in favor of current unstable.

something is broken in nixpkgs for zig 0.11 right now, which we need, so we can't do that yet.

---

You'll halso have to change the `extra_files` entry in `release-please-config.json` to point to the package.nix file instead of flake.nix file

File: flake.nix
Comment (adamcstephens) [reply flake.nix:26]: I left this in because two erlangs have to be compiled for this config, both 26 and 27. https://github.com/NixOS/nixpkgs/issues/322168

I'm happy to remove it, but anybody building will have to wait on the slow rebar tests to run.

Comment (mhanberg) [reply flake.nix:26]: I think that happens anyway, when i pulled down the branch and my direnv starting booting up the nix develop shell, it compiled otp 26 and rebar

Comment (adamcstephens) [reply flake.nix:26]: Yeah we can't avoid compiling for now, but running the *tests* for rebar add significant time to the build. I'll pull it out since most users will be using the cache anyway.

Comment (mhanberg) [reply flake.nix:26]: Does extending it remove the tests?

Comment (adamcstephens) [reply flake.nix:26]: Yes. It's effectively an override for these nested packages allowing the tests to be disabled.

Comment (mhanberg) [reply flake.nix:26]: oh, is that what the `doCheck = false` is doing? I had copied that from an example previously and didn't pay it much attention.

Comment (adamcstephens) [reply flake.nix:26]: Yes :)

**Review by mhanberg — APPROVED** [review]
Just make the change to the release please config and looks good, thanks!

**Comment by adamcstephens** [comment]
I think this should be good to go, unless you still wanted me to remove the rebar overrides.

**Comment by mhanberg** [comment]
I checked out the branch locally and as my shell was booting up nix develop, its running the the rebar tests.

still better than before, will merge once i get a `nix build` passing locally

**Pull request merged**
